### ISSUE TITLE
Force Charge triggering offputting

### ISSUE THREAD

**Issue by shieldgenerator7** [shieldgenerator7/Stoned_Unity#369]
The Force Charge still isn't quite right. It triggers when I don't want it to, usually when I click far into a wall where there are no blastables and it explodes against the wall anyway.

the main reason i detected the closest wall was to keep it blasting walls that were surrounded by hidden areas. Maybe the blast should land on the closest edge of the closest hidden area instead?

**Comment by shieldgenerator7** [comment]
Suggestion from Charlie:

My two cents is to make it act more like physical movement. Make the unicorn smoothly move to the location.
In pseudo code
1.	User clicks the dash target
2.	Disable user input
3.	Unicorn smoothly moves to the target over ~.5 seconds
1.	Unicorn RB turns to kinematic mode
2.	Set unicorn RB mass to some large number ~100x most in-game objects
3.	Every frame, call rb.move to the destination on the next frame (this will make sure physics is calculated correctly)
4.	If the path is blocked by indestructible walls, break lerp
5.	If the unicorn is damaged, break lerp
6.	Wait one fixed update frame
7.	Go back to 3.3
4.	Restore unicorn RB settings
5.	Set unicorn velocity to a fraction of what it would have been at the end of the frame. The amount of velocity kept increases as you 'charge up' the power.
6.	Set unicorn angular velocity to zero
7.	Wait one fixed update frame
8.	Enable user input
9.	Resume game

**Comment by shieldgenerator7** [comment]
Other options:
-make another way for the Force Charge to make an explosion
-remove the explosion altogether, meaning you move objects by teleporting past them and you destroy objects by using your own momentum to crash into them

**Comment by shieldgenerator7** [comment]
Option:
-Make it explode when you tap in the red area that is expanded by the Force Charge's charge

**Comment by shieldgenerator7** [comment]
Made some new changes that should fix this issue, but I have to test it with testers before I can confirm

**Comment by shieldgenerator7** [comment]
Working on it, trying to improve it with Camera Controller changes

**Comment by shieldgenerator7** [comment]
I think it has been fixed. Based on my own testing, this issue has been resolved.
Will reopen / open a new issue if testers find this difficult
